components/Weather.js

```diff
@@ -1,146 +1,221 @@
-            phoneLocated: true,
-          });
+          const { latitude, longitude } = position.coords;
+          this.getWeatherByCoords(latitude, longitude);
         },
         (error) => {
-          // See error code charts below.
           console.log(error.code, error.message);
         },
-        {enableHighAccuracy: true, timeout: 15000},
+        { enableHighAccuracy: true, timeout: 15000 },
       );
     }
   };
-  _parseWeatherData = (data) => {
-    const weather = {};
-    weather.city = datas.city.name;
-    weather.dataPerHour = [];
-    for (const data of datas.list) {
-      const perHour = {
-        temp: parseInt(data.main.temp),
-        humidity: data.main.humidity,
-        description: data.weather[0].description,
-        wind: data.wind.speed,
-        hour: data.dt_txt.substr(-8, 5),
-        date: data.dt,
-        icon: data.weather[0].icon,
+
+  extractWeatherData = (data) => {
+    const weatherPerHours = [];
+    data.forEach((weatherLog) => {
+      const hour = {
+        temp: parseInt(weatherLog.main.temp, 10),
+        humidity: weatherLog.main.humidity,
+        description: weatherLog.weather[0].description,
+        wind: weatherLog.wind.speed,
+        hour: weatherLog.dt_txt.substr(-8, 5),
+        icon: weatherLog.weather[0].icon,
+        dayNumber: new Date(weatherLog.dt * 1000).getDay(),
       };
-      weather.dataPerHour.push(perHour);
+      weatherPerHours.push(hour);
+    });
+    return weatherPerHours;
+  };
+
+  savePersistentData = async (key = '', data = '') => {
+    const { storageKey } = this;
+    try {
+      await AsyncStorage.setItem(`${storageKey}:${key}`, data);
+    } catch (e) {
+      console.error(`AsyncStorage error: ${e.message}`);
+    }
+  };
+
+  getPersistentData = async (key) => {
+    const { storageKey } = this;
+    try {
+      const value = await AsyncStorage.getItem(`${storageKey}:${key}`);
+      if (value !== null) {
+        return value;
+      }
+    } catch (e) {
+      console.error(`AsyncStorage error: ${e.message}`);
     }
+    return null;
+  };
+
+  parseWeatherData = (data) => {
+    const weather = {};
+    const days = ['NIEDZ.', 'PON.', 'WT.', 'ŚR.', 'CZW.', 'PT.', 'SOB.'];
+    weather.city = data.city.name;
+    weather.days = [];
+    const extractedData = this.extractWeatherData(data.list);
+    [weather.currenWeather] = extractedData;
+    weather.nearestHours = extractedData.slice(1, 5);
+    let daysCounter = 0;
+    for (let i = 0; i < extractedData.length; i += 1) {
+      if (extractedData[i].hour === '00:00') {
+        const day = {};
+        day.maxTemp = extractedData[i + 5].temp;
+        day.minTemp = extractedData[i + 9].temp;
+        day.iconDay = extractedData[i + 5].icon;
+        day.iconNight = extractedData[i + 9].icon;
+        daysCounter += 1;
+        day.name = days[extractedData[i + 5].dayNumber];
+        weather.days.push(day);
+      }
+      if (daysCounter === 4) {
+        break;
+      }
+    }
```

Comment:
I can be done better using array api. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array